base/operators.jl

```diff
@@ -961,6 +961,7 @@ const => = Pair
 start(p::Pair) = 1
 done(p::Pair, i) = i>2
 next(p::Pair, i) = (getfield(p,i), i+1)
+eltype{A,B}(p::Pair{A,B}) = Union{A,B}
```

Comment:
Aren't there also some fast paths in the dynamic dispatch for `Union`s with only a few elements?